shadowsocks/packet.go

```diff
@@ -27,6 +27,28 @@ var ErrShortPacket = errors.New("short packet")
 // This array must be at least service.maxNonceSize bytes.
 var zeroNonce [12]byte
 
+// Pack encrypts a Shadowsocks-UDP packet and returns a slice containing the encrypted packet.
+// dst must be big enough to hold the encrypted packet.
+// If plaintext and dst overlap but are not aligned for in-place encryption, this
+// function will panic.
+func Pack(dst, plaintext []byte, cipher shadowaead.Cipher) ([]byte, error) {
+	saltSize := cipher.SaltSize()
+	salt := dst[:saltSize]
```

Comment:
This may cause a buffer overflow if len(dst) < saltSize
We should check the buffer length first.

Reply:
Done